packages/gatsby/src/services/run-mutation-batch.ts

```diff
@@ -0,0 +1,26 @@
+import { IMutationAction } from "../state-machines/data-layer/types"
+import { Store, AnyAction } from "redux"
+import { IGatsbyState } from "../redux/types"
+import { IWaitingContext } from "../state-machines/waiting/types"
+import { assertStore } from "../utils/assert-store"
+import { actions } from "../redux/actions"
+
+const callRealApi = async (
+  event: IMutationAction,
+  store?: Store<IGatsbyState, AnyAction>
+): Promise<void> => {
+  assertStore(store)
+  const { type, payload } = event
+  if (type in actions) {
+    store.dispatch(actions[type](...payload))
+  }
+}
+
+export const runMutationBatch = ({
+  runningBatch = [],
+  store,
```

Comment:
I'm confused why TS does not require you to annotate the params of an exported function :s

Reply:
I think we only require types for returns and not args right now: https://github.com/gatsbyjs/gatsby/blob/425fdac890ac5260edb7419635f97bec1e58871b/.eslintrc.js#L142-L144

Don't see rule for args, unless there is one in "recommended" rules which we merge with our custom ones

Reply:
The params are explicitly typed. See line 22.

Reply:
:facepalm: